diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index 96c6db46c6..e53918a1b7 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -242,6 +242,11 @@
 #endif
 #endif
 
+; Camoufox specific files
+@RESPATH@/defaults/pref/local-settings.js
+@RESPATH@/distribution/policies.json
+@RESPATH@/camoufox.cfg
+
 ; [Default Preferences]
 ; All the pref files must be part of base to prevent migration bugs
 @RESPATH@/browser/@PREF_DIR@/firefox.js
diff --git a/lw/moz.build b/lw/moz.build
index e69de29bb2..77874d877b 100644
--- a/lw/moz.build
+++ b/lw/moz.build
@@ -0,0 +1,14 @@
+FINAL_TARGET_FILES += [
+  "camoufox.cfg",
+  "chrome.css",
+  "properties.json"
+]
+
+FINAL_TARGET_FILES.distribution += [
+  "policies.json",
+]
+
+# local-settings does not yet end up being pacakged.
+FINAL_TARGET_FILES.defaults.pref += [
+  "local-settings.js",
+]
diff --git a/moz.build b/moz.build
index 6129aa819d..3c1d77c638 100644
--- a/moz.build
+++ b/moz.build
@@ -237,0 +238,2 @@ include("build/templates.mozbuild")
+
+DIRS += ["lw"]
